srml/executive/src/lib.rs

```diff
@@ -213,6 +214,50 @@ impl<
 		header.state_root().check_equal(&storage_root);
 		assert!(header.state_root() == &storage_root, "Storage root must match that calculated.");
 	}
+
+	/// Check a given transaction for validity. This doesn't execute any
+	/// side-effects; it merely checks whether the transaction would panic if it were included or not.
+	/// 
+	/// Changes made to the storage should be discarded.
+	pub fn validate_transaction(uxt: Block::Extrinsic) -> TransactionValidity {
```

Comment:
Might be easier to have something like `Result<..>` here and use `?` syntax to early exit. Then maybe `From<Result<..>>` for `TransactionValidity`